library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -454,6 +509,14 @@ public long freeMemory(long windowId) throws IOException
         entryIter.remove();
 
         for (Map.Entry<Slice, BucketedValue> entry : windowData.entrySet()) {
+          /**
+           * The data still in memory and reachable before the memory released
+           * So put key into bloom filter here
+           */
+          if (bloomFilter != null) {
```

Comment:
The flag will be put when read data from files.
Let assume everything is in memory, do we still need bloom filter? probably not as it's simple a map. So I think filter only applies to decrease the chance of unnecessary reading from files